src/snapred/backend/recipe/algorithm/AlignAndFocusReductionAlgorithm.py

```diff
@@ -47,9 +47,23 @@ def PyExec(self):
         )
 
         self.mantidSnapper.executeQueue()
-        DMin = reductionIngredients.reductionState.stateConfig.focusGroups[0].dMin
-        DMax = reductionIngredients.reductionState.stateConfig.focusGroups[0].dMax
-        DeltaRagged = reductionIngredients.reductionState.stateConfig.focusGroups[0].dBin
+
+        DMin = []
+        DMax = []
+        DeltaRagged = []
+        dMin = {pgp.groupID: pgp.dResolution.minimum for pgp in reductionIngredients.pixelGroupingParameters}
+        dMax = {pgp.groupID: pgp.dResolution.maximum for pgp in reductionIngredients.pixelGroupingParameters}
+        dBin = {
+            pgp.groupID: pgp.dRelativeResolution / reductionIngredients.reductionState.instrumentConfig.NBins
+            for pgp in reductionIngredients.pixelGroupingParameters
+        }
+
+        groupIDs = [pgp.groupID for pgp in reductionIngredients.pixelGroupingParameters]
+        groupIDs.sort()
+        for i in groupIDs:
+            DMin.append(dMin[i])
+            DMax.append(dMax[i])
+            DeltaRagged.append(dBin[i])
```

Comment:
``` suggestion
        DMin = [dMin[groupID] for groupID in groupIDs]
        DMax = [dMax[groupID] for groupID in groupIDs]
        DeltaRagged = [dBin[groupID] for groupID in groupIDs]
```

Reply:
Or even
``` suggestion
        DMin = dMin.values()
        DMax = dMax.values()
        DeltaRagged = dBin.values()
```

Reply:
I'll do the first suggestion. If I did the second one, I don't think the lists will be sorted in ascending order